quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs

```diff
@@ -461,6 +505,109 @@ fn extract_json_val(
     }
 }
 
+/// extract a subfield from a TantivyValue. The path must be non-empty
+fn extract_val_from_tantivy_val(
+    full_path: &[&str],
+    tantivy_value: &mut [TantivyValue],
+) -> Vec<TantivyValue> {
```

Comment:
this looks like a good candidate for unit tests (e.g you will be able to address the TODO later without fear of a regression)